dcpquery/etl/__init__.py

```diff
@@ -36,21 +32,83 @@ def transform_bundle(bundle_uuid, bundle_version, bundle_path, bundle_manifest_p
                                             content_type=fm["content-type"],
                                             name=fm["name"],
                                             size=fm["size"]))
+
+    combine_file_data(result)
     return result
 
 
-def load_bundle(bundle, extractor, transformer):
-    bf_links = []
-    bundle_row = Bundle(uuid=bundle["uuid"], version=bundle["version"], manifest=bundle["manifest"])
-    for f in bundle["files"]:
-        filename = f.pop("name")
-        if filename == "links.json":
-            links = f['body']['links']
-        file_row = File(**f)
-        bf_links.append(BundleFileLink(bundle=bundle_row, file=file_row, name=filename))
-    config.db_session.add_all(bf_links)
-    config.db_session.commit()
-    load_links(links)
+def combine_file_data(bundle):
+    combined_file_data = {}
+    for f in bundle["manifest"]['files']:
+        uuid = f['uuid']
+        combined_file_data[uuid] = f
+
+    for f in bundle['files']:
+        uuid = f['uuid']
+        combined_file_data[uuid] = {**f, **combined_file_data[uuid]}
+    for uuid in combined_file_data:
+        file = combined_file_data[uuid]
+        filename = file['name']
+        regex_search_result = re.search('^(?:(?:.){1,}?)((?:[.][a-z]{1,10}){1,2})$', filename)
+        if regex_search_result:
+            file_extension = regex_search_result.group(1)
```

Comment:
What happens when there is no search result? It looks like this would crash if `file_extension` is not declared, or erroneously use a `file_extension` value from the previous file.

In this situation it's best to factor this out into a separate function like `get_extension_for_file()`.

Consider adding a unit test for this function where the first file has no extension.

Reply:
👍